src/tools/tidy/src/style.rs

```diff
@@ -306,6 +308,21 @@ pub fn check(path: &Path, bad: &mut bool) {
             let mut err = |msg: &str| {
                 tidy_error!(bad, "{}:{}: {}", file.display(), i + 1, msg);
             };
+
+            if trimmed.contains("dbg!")
+                && !trimmed.starts_with("//")
```

Comment:
what if there is a `gdb!` in multiple comments part?
for example:

```rust
/* .....
   blah  gdb!
*/
```

Reply:
I hope you mean dbg! not gdb! :)

Also yes, this has already been noted in [this comment.](https://github.com/rust-lang/rust/pull/106327#issuecomment-1368238963)

Reply:
ah, yes, I think we need to parse the source files, otherwise there will be some false warning.
But seems our current checking are all without parsing, so it may not a issue in practice...

Reply:
This is a suppressable lint for a reason. :)